lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb

```diff
@@ -6,35 +6,42 @@ module FirebaseAppDistributionAuthClient
       TOKEN_CREDENTIAL_URI = "https://oauth2.googleapis.com/token"
 
       def fetch_auth_token(google_service_path)
-        if ENV["XDG_CONFIG_HOME"]
-          config_path = File.expand_path("configstore/firebase-tools.json", ENV["XDG_CONFIG_HOME"])
-        else
-          config_path = File.expand_path(".config/configstore/firebase-tools.json", "~")
-        end
+        config_path = find_firebase_tools
         if !google_service_path.nil? && !google_service_path.empty?
           service_account(google_service_path)
         elsif ENV["FIREBASE_TOKEN"]
           firebase_token(ENV["FIREBASE_TOKEN"])
         elsif ENV["GOOGLE_APPLICATION_CREDENTIALS"]
           service_account(ENV["GOOGLE_APPLICATION_CREDENTIALS"])
-        elsif File.exist?(config_path)
-          begin
-            refresh_token = JSON.parse(File.read(config_path))['tokens']['refresh_token']
-            if refresh_token.nil? || refresh_token.empty?
-              UI.crash!(ErrorMessage::MISSING_REFRESH_TOKEN)
-            else
-              firebase_token(refresh_token)
-            end
-          rescue NoMethodError
-            UI.crash!(ErrorMessage::MISSING_REFRESH_TOKEN)
-          end
+        elsif (refresh_token = has_refresh_token(config_path))
+          firebase_token(refresh_token)
         else
           UI.crash!(ErrorMessage::MISSING_CREDENTIALS)
         end
       end
 
       private
 
+      def has_refresh_token(config_path)
+        if File.exist?(config_path)
+          begin
+           refresh_token = JSON.parse(File.read(config_path))['tokens']['refresh_token']
+           unless refresh_token.nil? && refresh_token.empty?
+             refresh_token
+           end
+         rescue NoMethodError
```

Comment:
If you're going to have an empty rescue block, I think a comment is warranted to describe when this case will happen, and why it's OK for it to be empty.